bitrise/util.go

```diff
@@ -328,357 +309,3 @@ func ReadSpecStep(pth string) (stepmanModels.StepModel, error) {
-			} else {
-				hasOptions = true
-			}
-
-			if wfOptions.Title != nil && sOptions.Title != nil && *wfOptions.Title == *sOptions.Title {
-				wfOptions.Title = nil
-			} else {
-				hasOptions = true
-			}
-
-			if wfOptions.Description != nil && sOptions.Description != nil && *wfOptions.Description == *sOptions.Description {
-				wfOptions.Description = nil
-			} else {
-				hasOptions = true
-			}
-
-			if wfOptions.Summary != nil && sOptions.Summary != nil && *wfOptions.Summary == *sOptions.Summary {
-				wfOptions.Summary = nil
-			} else {
-				hasOptions = true
-			}
-
-			if wfOptions.Category != nil && sOptions.Category != nil && *wfOptions.Category == *sOptions.Category {
-				wfOptions.Category = nil
-			} else {
-				hasOptions = true
-			}
-
-			if isStringSliceWithSameElements(wfOptions.ValueOptions, sOptions.ValueOptions) {
-				wfOptions.ValueOptions = []string{}
-			} else {
-				hasOptions = true
-			}
-
-			if wfOptions.IsRequired != nil && sOptions.IsRequired != nil && *wfOptions.IsRequired == *sOptions.IsRequired {
-				wfOptions.IsRequired = nil
-			} else {
-				hasOptions = true
-			}
-
-			if wfOptions.IsDontChangeValue != nil && sOptions.IsDontChangeValue != nil && *wfOptions.IsDontChangeValue == *sOptions.IsDontChangeValue {
-				wfOptions.IsDontChangeValue = nil
-			} else {
-				hasOptions = true
-			}
-
-			if wfOptions.IsTemplate != nil && sOptions.IsTemplate != nil && *wfOptions.IsTemplate == *sOptions.IsTemplate {
-				wfOptions.IsTemplate = nil
-			} else {
-				hasOptions = true
-			}
-
-			if !hasOptions && sameValue {
-				// default env
-			} else {
-				if hasOptions {
-					input[envmanModels.OptionsKey] = wfOptions
-				} else {
-					delete(input, envmanModels.OptionsKey)
-				}
-
-				inputs = append(inputs, input)
-			}
-		}
-
-		workflowStep.Inputs = inputs
-
-		// We need only key-value and title from spec outputs
-		outputs := []envmanModels.EnvironmentItemModel{}
-		for _, output := range specStep.Outputs {
-			sKey, sValue, err := output.GetKeyValuePair()
-			if err != nil {
-				return err
-			}
-
-			sOptions, err := output.GetOptions()
-			if err != nil {
-				return err
-			}
-
-			newOutput := envmanModels.EnvironmentItemModel{
-				sKey: sValue,
-				envmanModels.OptionsKey: envmanModels.EnvironmentItemOptionsModel{
-					Title: sOptions.Title,
-				},
-			}
-
-			outputs = append(outputs, newOutput)
-		}
-
-		workflowStep.Outputs = outputs
-
-		(*stepListItem)[compositeStepIDStr] = workflowStep
-	}
-
-	// Cleanup
-	if err := command.RemoveDir(tempStepCloneDirPath); err != nil {
-		return errors.New(fmt.Sprint("Failed to remove step clone dir: ", err))
-	}
-	if err := command.RemoveDir(tempStepYMLDirPath); err != nil {
-		return errors.New(fmt.Sprint("Failed to remove step clone dir: ", err))
-	}
-
-	return nil
-}
-
-// RemoveConfigRedundantFieldsAndFillStepOutputs ...
-func RemoveConfigRedundantFieldsAndFillStepOutputs(config *models.BitriseDataModel) error {
```

Comment:
Public method, but appears unused: https://github.com/search?q=org:bitrise-io+RemoveConfigRedundantFieldsAndFillStepOutputs+NOT+repo:bitrise-io/bitrise&type=code